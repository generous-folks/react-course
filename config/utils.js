const getExerciseFromProcess = () => {
  const exNumber = process.argv[2];
  if(!process.env.APP_EXERCISE || !exNumber) {
    return null;
  }

  return exNumber;
}

module.exports = getExerciseFromProcess;
