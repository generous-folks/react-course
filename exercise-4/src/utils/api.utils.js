const endpoint = '/fixtures/articles.json';

export const getArticles = () =>
  fetch(endpoint)
    .then(res => res.json())
    .catch(err => console.log(err));
