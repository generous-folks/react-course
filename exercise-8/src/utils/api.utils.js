const endpoint = '/fixtures/articles.json';

export const getArticles = () =>
  fetch(endpoint)
    .then(res => res.json())
    .catch(err => console.log(err));

export const signIn = () => Promise.resolve({ id: 'xyz', mail: 'foo@bar.com', name: 'Foo Bar' });
export const signOut = args => Promise.resolve(args);
