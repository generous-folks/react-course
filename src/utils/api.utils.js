const endpoint = 'https://us-central1-gofire-1280e.cloudfunctions.net/ArticleAPI';

export const getArticles = () =>
  fetch(endpoint)
    .then(res => res.json())
    .catch(err => console.log(err));

export const setArticle = article =>
  fetch(endpoint, { method: 'POST', body: article })
    .then(res => res.json())
    .catch(err => console.log(err));

export const updateArticle = article =>
  fetch(endpoint, { method: 'PUT', body: article })
    .then(res => res.json())
    .catch(err => console.log(err));

export const deleteArticle = article =>
  fetch(endpoint, { method: 'DELETE', body: article })
    .then(res => res.json())
    .catch(err => console.log(err));
