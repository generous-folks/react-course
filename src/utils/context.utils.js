export const dispatchThunk = dispatch => param => {
  if (typeof param === 'function') {
    return param(dispatch);
  }

  return dispatch(param);
}
