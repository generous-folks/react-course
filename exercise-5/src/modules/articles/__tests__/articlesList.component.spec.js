import React, { useEffect, useState } from 'react';

import { shallow } from 'enzyme';

import { Grid } from '@material-ui/core';

import { ArticlesList } from '../articlesList.component';

import MockedArticles from '../../../../../fixtures/articles.json';
import * as ApiUtils from '../../../utils/api.utils';

let wrapper;
const emptyArray = [];

ApiUtils.getArticles = jest.fn().mockResolvedValue(MockedArticles);

jest.mock('react', () => global.mockReactWithHooks({ effect: true, state: true }));

const getWrapper = () => shallow(<ArticlesList />);

beforeEach(() => {
  jest.clearAllMocks();
  useEffect.mockClear();
  useState.mockClear();
  wrapper = getWrapper();
});

describe('<ArticlesList />', () => {
  describe('Snapshot', () => {
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should render list correctly', () => {
      useState.mockReturnValueOnce([MockedArticles, jest.fn()]);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Markup checks', () => {
    it('should contain the correct markup', () => {
      expect(wrapper.find(Grid).exists()).toBeTruthy();
    });
  });

  describe('State checks', () => {
    // Don't test Jest mocks or React, it is useless
    // In the following example we test that our react mock is working
    // const [initialsArticles] = useState.mock.results[0].value;
    // expect(initialsArticles).toEqual(emptyArray);

    it('should call useState with an empty array', () => {
      expect(useState).toHaveBeenCalledWith(emptyArray);
    });
  });

  describe('Effects checks', () => {
    it('should call useEffect with a function and an empty array', () => {
      expect(useEffect).toHaveBeenCalledWith(expect.any(Function), [emptyArray]);
    });

    it('should call getArticles once', () => {
      expect(ApiUtils.getArticles).toHaveBeenCalledTimes(1);
    });

    it('should call getArticles once even when re-rendered', () => {
      wrapper.update();
      expect(ApiUtils.getArticles).toHaveBeenCalledTimes(1);
    });

    it('should call setArticles once with MockedArticles', () => {
      const [, setArticles] = useState.mock.results[0].value;
      expect(setArticles).toHaveBeenCalledWith(MockedArticles);
      expect(setArticles).toHaveBeenCalledTimes(1);
    });
  });
});
