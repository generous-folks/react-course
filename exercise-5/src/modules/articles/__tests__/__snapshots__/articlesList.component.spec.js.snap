// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticlesList /> Snapshot should render correctly 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  spacing={4}
/>
`;

exports[`<ArticlesList /> Snapshot should render list correctly 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  spacing={4}
/>
`;
